packages/core/src/textures/utils/mapTypeAndFormatToInternalFormat.ts

```diff
@@ -0,0 +1,132 @@
+import { FORMATS, TYPES } from '@pixi/constants';
+
+/**
+ * Returns a lookup table that maps each type-format pair to a compatible internal format.
+ *
+ * @memberof PIXI
+ * @function mapTypeAndFormatToInternalFormat
+ * @private
+ * @param {WebGLRenderingContext} gl - The rendering context.
+ * @return {{ [type: number]: { [format: number]: number } }} Lookup table.
+ */
+export function mapTypeAndFormatToInternalFormat(gl: WebGLRenderingContextBase):
+    { [type: number]: { [format: number]: number } }
```

Comment:
Something like this might be a more specific format:
`{ [type: TYPES]: { [format: FORMATS]: number } }`

Reply:
I tried that first but it got the following error message:
> An index signature parameter type cannot be a union type. Consider using a mapped object type instead.

But it seems to compile without issues nonetheless. I'm going to change it.

Reply:
Not that critical, just seeing if we could do something more specific.

Reply:
The correct way for that would be:

```ts
Record<TYPES, Record<FORMATS, number>>
```

Reply:
I vaguely thought there were string-only key restrictions on Record template. But I think it supports numbers/symbols too.